src/NzbDrone.Core/ImportLists/TMDb/Company/TMDbCompanyRequestGenerator.cs

```diff
@@ -0,0 +1,59 @@
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using NLog;
+using NzbDrone.Common.Http;
+
+namespace NzbDrone.Core.ImportLists.TMDb.Company
+{
+    public class TMDbCompanyRequestGenerator : IImportListRequestGenerator
+    {
+        public TMDbCompanySettings Settings { get; set; }
+        public IHttpClient HttpClient { get; set; }
+        public IHttpRequestBuilderFactory RequestBuilder { get; set; }
+        public Logger Logger { get; set; }
+        public int MaxPages { get; set; }
+
+        public TMDbCompanyRequestGenerator()
+        {
+        }
+
+        public virtual ImportListPageableRequestChain GetMovies()
+        {
+            var pageableRequests = new ImportListPageableRequestChain();
+
+            pageableRequests.Add(GetMoviesRequest());
+
+            return pageableRequests;
+        }
+
+        private IEnumerable<ImportListRequest> GetMoviesRequest()
+        {
+            Logger.Info($"Importing TMDb movies from company: {Settings.CompanyId}");
+
+            var requestBuilder = RequestBuilder.Create()
+                                               .SetSegment("api", "3")
+                                               .SetSegment("route", "discover")
+                                               .SetSegment("id", $"movie?with_companies={Settings.CompanyId}")
+                                               .SetSegment("secondaryRoute", "");
+
+            var jsonResponse = JsonConvert.DeserializeObject<MovieSearchResource>(HttpClient.Execute(requestBuilder.Build()).Content);
+
+            MaxPages = jsonResponse.TotalPages;
+
+            for (var pageNumber = 1; pageNumber <= MaxPages; pageNumber++)
+            {
+                requestBuilder = RequestBuilder.Create()
```

Comment:
FYI requestBuilder.AddQueryParam adds it with a ? and it needs to be an & hence why it is like this

Reply:
You should be able to use add query parameters, you just need to use it for both with_companies and page. 

The company query parameter can be set before the loop, then set the page query parameter in the loop with the replace flag to true to ensure it gets overwritten on each request